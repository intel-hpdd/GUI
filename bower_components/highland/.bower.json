{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}