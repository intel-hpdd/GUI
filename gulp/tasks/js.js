//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
