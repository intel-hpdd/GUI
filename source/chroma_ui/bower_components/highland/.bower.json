{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}