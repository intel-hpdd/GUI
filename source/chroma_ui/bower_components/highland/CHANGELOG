Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
