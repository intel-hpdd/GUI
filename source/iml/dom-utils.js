// @flow

type QuerySelector<T: HTMLElement> = (HTMLElement | Document, string) => T;

export const querySelector: QuerySelector<*> = (
  elm: HTMLElement | Document,
  sel: string
) => {
  const selected = elm.querySelector(sel);

  if (!selected)
    throw new Error(
      `Could not find selector ${sel} for element ${elm.nodeName}`
    );

  return selected;
};
