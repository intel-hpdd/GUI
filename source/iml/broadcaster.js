// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import highland from 'highland';

import type {
  HighlandStreamT,
  errorWrapT
} from 'highland';


type streamFnT = () => HighlandStreamT<any>;

export default function broadcaster (source$:HighlandStreamT<any>):streamFnT {
  var latest:mixed;

  const viewers = [];

  source$
    .errors(error => {
      const err:errorWrapT = {
        __HighlandStreamError__: true,
        error
      };

      viewers.forEach(v => v.write(err));
    })
    .each((xs:mixed) => {
      latest = xs;

      viewers.forEach(v => v.write(xs));
    });

  const fn = () => {
    const viewer$:HighlandStreamT<any> = highland()
      .onDestroy(() => {
        const idx = viewers
          .indexOf(viewer$);

        if (idx !== -1)
          viewers.splice(idx, 1);
      });

    if (latest)
      viewer$.write(latest);

    viewers.push(viewer$);

    return viewer$;
  };

  fn.endBroadcast = () => source$.destroy();

  return fn;
}
