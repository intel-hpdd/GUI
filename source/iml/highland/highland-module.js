// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
