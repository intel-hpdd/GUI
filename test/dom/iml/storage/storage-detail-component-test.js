// @flow

import highland, { type HighlandStreamT } from "highland";
import broadcaster from "../../../../source/iml/broadcaster.js";
import type { StorageResource } from "../../../../source/iml/storage/storage-types.js";
import { renderToSnapshot } from "../../../test-utils.js";
import { querySelector } from "../../../../source/iml/dom-utils.js";

const createResourceData = (deletable: boolean): StorageResource => ({
  alerts: [{}],
  alias: "my-resource",
  attributes: {
    "1": {
      class: "my-class",
      label: "Name",
      markup: "<span>resource1</span>",
      raw: "resource1"
    }
  },
  charts: [
    {
      series: ["1", "2"],
      title: "chart title"
    }
  ],
  class_name: "my-class",
  default_alias: "my-resource",
  deletable,
  id: 1,
  parent_classes: ["class1", "class2"],
  plugin_name: "plugin-name",
  propagated_alerts: [{}],
  resource_uri: "/storage/resource/33",
  stats: {
    "1": {
      data: null,
      label: "World Series",
      name: "world-series",
      type: "timeseries",
      unit_name: "time-unit"
    },
    "2": {
      data: {
        bin_labels: ["label1"],
        values: [1, 2]
      },
      label: "Histogram",
      name: "Histogram",
      type: "histogram",
      unit_name: "time-unit"
    }
  },
  storage_id_str: "storage-id-str"
});

describe("storage detail component", () => {
  let StorageDetail,
    storageResource$: HighlandStreamT<StorageResource>,
    alertIndicator$: () => HighlandStreamT<Object>,
    mockStorageResourceHistogram,
    mockStorageResourceTimeSeries,
    mockSocketStream,
    mockGlobal,
    socket$: HighlandStreamT<Object>,
    node: HTMLElement,
    alias: HTMLInputElement,
    saveBtn: HTMLButtonElement,
    deleteBtn: HTMLButtonElement;

  beforeEach(() => {
    storageResource$ = highland();
    alertIndicator$ = broadcaster(highland([[]]));

    mockStorageResourceHistogram = jest.fn(() => "storageResourceHistogram");
    mockStorageResourceTimeSeries = jest.fn(() => "storageResourceTimeSeries");
    socket$ = highland();
    mockSocketStream = jest.fn(() => socket$);
    mockGlobal = {
      location: {
        href: "/storage/23"
      }
    };

    jest.mock("../../../../source/iml/storage/storage-resource-histogram.js", () => mockStorageResourceHistogram);

    jest.mock("../../../../source/iml/storage/storage-resource-time-series.js", () => mockStorageResourceTimeSeries);

    jest.mock("../../../../source/iml/socket/socket-stream.js", () => mockSocketStream);

    jest.mock("../../../../source/iml/global.js", () => mockGlobal);

    ({ StorageDetail } = require("../../../../source/iml/storage/storage-detail-component.js"));

    storageResource$.write(createResourceData(false));

    node = renderToSnapshot(<StorageDetail stream={storageResource$} alertIndicatorB={alertIndicator$} />);

    querySelector(document, "body").appendChild(node);

    saveBtn = (querySelector(node, "#saveButton"): any);
    alias = (querySelector(node, "#alias"): any);
  });

  afterEach(() => {
    if (document.body) document.body.removeChild(node);
  });

  it("should render the storage detail form", () => {
    expect(node).toMatchSnapshot();
  });

  describe("changing the alias", () => {
    beforeEach(() => {
      alias.value = "something different";
      alias.dispatchEvent(new Event("input"));
    });

    describe("saving the alias", () => {
      beforeEach(() => {
        saveBtn.click();
      });

      it("should update the storage resource", () => {
        expect(mockSocketStream).toHaveBeenCalledWith(
          "/storage_resource/1",
          {
            method: "put",
            json: {
              alias: "something different"
            }
          },
          true
        );
      });

      it("should match the snapshot", () => {
        expect(node).toMatchSnapshot();
      });

      describe("after saving the alias", () => {
        beforeEach(() => {
          socket$.end();
        });

        it("should have the new alias", () => {
          expect(alias.value).toEqual("something different");
        });

        it("should match the snapshot", () => {
          expect(node).toMatchSnapshot();
        });
      });
    });
  });

  describe("deleting a resource", () => {
    beforeEach(() => {
      storageResource$.write(createResourceData(true));
      deleteBtn = (querySelector(node, "#deleteButton"): any);

      deleteBtn.click();
    });

    it("should delete the resource", () => {
      expect(mockSocketStream).toHaveBeenCalledWith(
        "/storage_resource/1",
        {
          method: "delete"
        },
        true
      );
    });

    it("should match the snapshot", () => {
      expect(node).toMatchSnapshot();
    });

    describe("after deleting", () => {
      beforeEach(() => {
        socket$.end();
      });

      it("should navigate to /storage", () => {
        expect(mockGlobal.location.href).toEqual("/ui/configure/storage");
      });
    });
  });
});
