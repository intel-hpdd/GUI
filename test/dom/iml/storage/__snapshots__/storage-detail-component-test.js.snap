// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage detail component changing the alias saving the alias after saving the alias should match the snapshot 1`] = `
<storage-detail
  alert-indicator-b="alertIndicatorB"
  class="ng-scope ng-isolate-scope"
  storage-resource-$="storageResourceB"
>
  <div
    class="container container-full container"
  >
    <form
      autocomplete="off"
    >
      <div
        class="detail-panel form-group"
      >
        <h4
          class="section-header"
        >
          my-class
           Detail
        </h4>
        <div
          class="detail-row"
        >
          <div>
            Alias
          </div>
          <div>
            <input
              class="form-control"
              id="alias"
              name="alias"
              required=""
              type="text"
              value="my-resource"
            />
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            Name
            :
          </div>
          <div>
            <span>
              <span>
                resource1
              </span>
            </span>
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            Alerts:
          </div>
          <div>
            <span
              class="record-state"
            >
              <span
                class="icon-wrap tooltip-container tooltip-hover"
                popoverbutton="true"
              >
                <i
                  class="fa activate-popover fa-check-circle"
                />
                
                <div
                  class="tooltip inferno-tt top small "
                >
                  <div
                    class="tooltip-arrow"
                  />
                  <div
                    class="tooltip-inner"
                  >
                    <span>
                      No Issues
                    </span>
                  </div>
                </div>
              </span>
              
              <span
                class="state-label"
                style="padding-left: 10px;"
              >
                No Issues
              </span>
            </span>
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            
          </div>
          <div>
            <button
              class="btn btn-default"
              disabled=""
              id="saveButton"
              type="submit"
            >
              Save 
              
            </button>
          </div>
        </div>
      </div>
    </form>
    <div>
      <div>
        <h4
          class="text-center"
          style="margin-top: 50px;"
        >
          chart title
        </h4>
        storageResourceHistogram
      </div>
    </div>
  </div>
</storage-detail>
`;

exports[`storage detail component changing the alias saving the alias should match the snapshot 1`] = `
<storage-detail
  alert-indicator-b="alertIndicatorB"
  class="ng-scope ng-isolate-scope"
  storage-resource-$="storageResourceB"
>
  <div
    class="container container-full container"
  >
    <form
      autocomplete="off"
    >
      <div
        class="detail-panel form-group"
      >
        <h4
          class="section-header"
        >
          my-class
           Detail
        </h4>
        <div
          class="detail-row"
        >
          <div>
            Alias
          </div>
          <div>
            <input
              class="form-control"
              disabled=""
              id="alias"
              name="alias"
              required=""
              type="text"
              value="my-resource"
            />
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            Name
            :
          </div>
          <div>
            <span>
              <span>
                resource1
              </span>
            </span>
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            Alerts:
          </div>
          <div>
            <span
              class="record-state"
            >
              <span
                class="icon-wrap tooltip-container tooltip-hover"
                popoverbutton="true"
              >
                <i
                  class="fa activate-popover fa-check-circle"
                />
                
                <div
                  class="tooltip inferno-tt top small "
                >
                  <div
                    class="tooltip-arrow"
                  />
                  <div
                    class="tooltip-inner"
                  >
                    <span>
                      No Issues
                    </span>
                  </div>
                </div>
              </span>
              
              <span
                class="state-label"
                style="padding-left: 10px;"
              >
                No Issues
              </span>
            </span>
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            
          </div>
          <div>
            <button
              class="btn btn-default"
              disabled=""
              id="saveButton"
              type="submit"
            >
              Save 
              <i
                class="fa fa-spinner fa-spin"
              />
            </button>
          </div>
        </div>
      </div>
    </form>
    <div>
      <div>
        <h4
          class="text-center"
          style="margin-top: 50px;"
        >
          chart title
        </h4>
        storageResourceHistogram
      </div>
    </div>
  </div>
</storage-detail>
`;

exports[`storage detail component deleting a resource should match the snapshot 1`] = `
<storage-detail
  alert-indicator-b="alertIndicatorB"
  class="ng-scope ng-isolate-scope"
  storage-resource-$="storageResourceB"
>
  <div
    class="container container-full container"
  >
    <form
      autocomplete="off"
    >
      <div
        class="detail-panel form-group"
      >
        <h4
          class="section-header"
        >
          my-class
           Detail
        </h4>
        <div
          class="detail-row"
        >
          <div>
            Alias
          </div>
          <div>
            <input
              class="form-control"
              disabled=""
              id="alias"
              name="alias"
              required=""
              type="text"
              value="my-resource"
            />
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            Name
            :
          </div>
          <div>
            <span>
              <span>
                resource1
              </span>
            </span>
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            Alerts:
          </div>
          <div>
            <span
              class="record-state"
            >
              <span
                class="icon-wrap tooltip-container tooltip-hover"
                popoverbutton="true"
              >
                <i
                  class="fa activate-popover fa-check-circle"
                />
                
                <div
                  class="tooltip inferno-tt top small "
                >
                  <div
                    class="tooltip-arrow"
                  />
                  <div
                    class="tooltip-inner"
                  >
                    <span>
                      No Issues
                    </span>
                  </div>
                </div>
              </span>
              
              <span
                class="state-label"
                style="padding-left: 10px;"
              >
                No Issues
              </span>
            </span>
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            <button
              class="btn btn-danger"
              disabled=""
              id="deleteButton"
              type="button"
            >
              Delete 
              <i
                class="fa fa-spinner fa-spin"
              />
            </button>
          </div>
          <div>
            <button
              class="btn btn-default"
              disabled=""
              id="saveButton"
              type="submit"
            >
              Save 
              <i
                class="fa fa-spinner fa-spin"
              />
            </button>
          </div>
        </div>
      </div>
    </form>
    <div>
      <div>
        <h4
          class="text-center"
          style="margin-top: 50px;"
        >
          chart title
        </h4>
        storageResourceHistogram
      </div>
    </div>
  </div>
</storage-detail>
`;

exports[`storage detail component should render the storage detail form 1`] = `
<storage-detail
  alert-indicator-b="alertIndicatorB"
  class="ng-scope ng-isolate-scope"
  storage-resource-$="storageResourceB"
>
  <div
    class="container container-full container"
  >
    <form
      autocomplete="off"
    >
      <div
        class="detail-panel form-group"
      >
        <h4
          class="section-header"
        >
          my-class
           Detail
        </h4>
        <div
          class="detail-row"
        >
          <div>
            Alias
          </div>
          <div>
            <input
              class="form-control"
              id="alias"
              name="alias"
              required=""
              type="text"
              value="my-resource"
            />
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            Name
            :
          </div>
          <div>
            <span>
              <span>
                resource1
              </span>
            </span>
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            Alerts:
          </div>
          <div>
            <span
              class="record-state"
            >
              <span
                class="icon-wrap tooltip-container tooltip-hover"
                popoverbutton="true"
              >
                <i
                  class="fa activate-popover fa-check-circle"
                />
                
                <div
                  class="tooltip inferno-tt top small "
                >
                  <div
                    class="tooltip-arrow"
                  />
                  <div
                    class="tooltip-inner"
                  >
                    <span>
                      No Issues
                    </span>
                  </div>
                </div>
              </span>
              
              <span
                class="state-label"
                style="padding-left: 10px;"
              >
                No Issues
              </span>
            </span>
          </div>
        </div>
        <div
          class="detail-row"
        >
          <div>
            
          </div>
          <div>
            <button
              class="btn btn-default"
              disabled=""
              id="saveButton"
              type="submit"
            >
              Save 
              
            </button>
          </div>
        </div>
      </div>
    </form>
    <div>
      <div>
        <h4
          class="text-center"
          style="margin-top: 50px;"
        >
          chart title
        </h4>
        storageResourceHistogram
      </div>
    </div>
  </div>
</storage-detail>
`;
